### ISSUE TITLE
How to Mock IorganiationService.Retrieve & RetrieveMultiple methods

### ISSUE THREAD

**Issue by krishire** [jordimontana82/fake-xrm-easy#68]
I have started using FakeXrmEasy but struggling to Mock the retrieve method of the FakedOrganisationService object. Earlier I was using in this fashion base.OrganizationServiceMock.Setup(t =>
                t.Retrieve(It.Is<string>(s => s == "systemuser"),
               It.IsAny<Guid>(),  
               It.IsAny<Microsoft.Xrm.Sdk.Query.ColumnSet>())) // don't care about a specific ColumnSet
               .Returns(activeUserNameMock);
Please provide some sample on how to implement the proxy retrieve method.

Basically, In my actual plugin code, I'm retrieving the SystemUser entity for an user in this fashion:
Entity userName = service.Retrieve("systemuser", userId, new Microsoft.Xrm.Sdk.Query.ColumnSet(new[] { "domainname", "fullname" }));

**Comment by jordimontana82** [comment]
Hi @krishire 

You don't need to mock those methods, there are already "mocked", which is the point of the framework. 

The only thing you need to do is to get an instance of a fake organization service like this

    var context = new XrmFakedContext();
    var service = context.GetFakedOrganizationService();

    //Then setup your initial state with a single system user, for example
    context.Initialize(new List<Entity>() {
        new SystemUser() { Id = Guid.NewGuid(), DomainName = "Domain", Fullname = "Some name" };
    //After this, any call to service.Retrieve or service.RetrieveMultiple will return that user provided the criteria is met.

More examples [here](https://github.com/jordimontana82/fake-xrm-easy/blob/master/FakeXrmEasy.Tests.Shared/FakeContextTests/FakeContextTestRetrieve.cs) and [here](https://github.com/jordimontana82/fake-xrm-easy/blob/master/FakeXrmEasy.Tests/FakeContextTests/LinqTests/FakeContextTestLinqQueries.cs)

**Comment by krishire** [comment]
Many Thanks Jordi, just realised that, thanks for your immediately reply.

**Comment by jordimontana82** [comment]
No worries

**Comment by krishire** [comment]
fakedContext.GetDefaultPluginContext().InitiatingUserId and PluginExecutionContext.InitiatingUserId within the plugin code are not matching, any idea.

**Comment by jordimontana82** [comment]
Hi @krishire 

Could you please provide a unit test to reproduce this?

From the GetDefaultPluginContext [implementation](https://github.com/jordimontana82/fake-xrm-easy/blob/274e87260d6f9fb416a754c4a3db1c1f7ab1745d/FakeXrmEasy.Shared/XrmFakedContext.Plugins.cs#L48), it should retrieve the one you passed to the context, or, a default one, if it was empty

**Comment by krishire** [comment]
Thanks a  lot Again Jordi, found the issue in my code,  re-initialised the initiatingUserId in the context, after creating the required base record using the default plugin context's intiatingUserId value.

**Comment by jordimontana82** [comment]
Grand! You're welcome.
